packages/social/src/abstract-provider.service.ts

```diff
@@ -155,11 +172,47 @@ export abstract class AbstractProvider<AuthParameters extends ObjectType, UserIn
    */
   protected readonly scopeSeparator: string = ' ';
 
+  /**
+   * Property used to enable code flow with PCKE.
+   *
+   * @protected
+   * @type {boolean}
+   * @memberof AbstractProvider
+   */
+   protected readonly useCodeVerifier: boolean = false;
+
+  /**
+   * Property use Plain Method within code challenge on PCKE.
+   *
+   * @protected
+   * @type {boolean}
+   * @memberof AbstractProvider
+   */
+   protected readonly codeChallengeMethodPlain: boolean = false;
+
+  /**
+   * Property use Plain Method within code challenge on PCKE.
+   *
+   * @protected
+   * @type {boolean}
+   * @memberof AbstractProvider
+   */
+   protected readonly codeChallengeSecretPath: string = 'settings.social.secret.codeChallengeSecret';
```

Comment:
Should we rename the `codeChallengeSecret` config key to `codeVerifierSecret` as well as it encrypts the code verifier and not its hash?

Reply:
done